src/test/java/com/microsoft/sqlserver/testframework/DBResultSet.java

```diff
@@ -85,4 +117,325 @@ public Object getObject(int index) throws SQLException {
+                                    + " ,received: " + (((Timestamp) retrieved).getTime()));
+                    break;
+                }
+                else
+                    assertTrue(("" + Timestamp.valueOf((LocalDateTime) expectedData)).equalsIgnoreCase("" + retrieved), "Unexpected datetime2 value, "
+                            + "expected: " + Timestamp.valueOf((LocalDateTime) expectedData) + " ,received: " + retrieved);
+                break;
+
+            case java.sql.Types.DATE:
+                assertTrue((("" + expectedData).equalsIgnoreCase("" + retrieved)),
+                        "Unexpected date value, expected: " + expectedData + " ,received: " + retrieved);
+                break;
+
+            case java.sql.Types.TIME:
+                assertTrue(("" + Time.valueOf((LocalTime) expectedData)).equalsIgnoreCase("" + retrieved),
+                        "Unexpected time value, exptected: " + Time.valueOf((LocalTime) expectedData) + " ,received: " + retrieved);
+                break;
+
+            case microsoft.sql.Types.DATETIMEOFFSET:
+                assertTrue(("" + expectedData).equals("" + retrieved),
+                        " unexpected DATETIMEOFFSET value, expected: " + expectedData + " ,received: " + retrieved);
+                break;
+
+            case java.sql.Types.BINARY:
+                assertTrue(parseByte((byte[])expectedData, (byte[])retrieved),
+                        " unexpected BINARY value, expected: " + expectedData + " ,received: " + retrieved);
+                break;
+
+            case java.sql.Types.VARBINARY:
+                assertTrue(parseByte((byte[])expectedData, (byte[])retrieved),
+                        " unexpected BINARY value, expected: " + expectedData + " ,received: " + retrieved);
+                break;
+            default:
+                fail("Unhandled JDBCType " + JDBCType.valueOf(metaData.getColumnType(ordinal + 1)));
+                break;
+        }
+    }
+    
+    private  boolean parseByte(byte[] expectedData, byte[] retrieved) {
+        for (int j = 0; j < expectedData.length; j++) {
+            assertTrue(
+                    expectedData[j] == retrieved[j] && expectedData[j] == retrieved[j] && expectedData[j] == retrieved[j],
+                            " unexpected BINARY value, expected");
+        }
+        return true;
+    }
+
+    private Object getXXX(int idx, Class coercion) throws SQLException {
+        if (coercion == Object.class) {
+            return this.getObject(idx);
+        }
+        return null;
+    }
+
+    /**
+     * 
+     * @return
+     */
+    public DBResultSetMetaData getMetaData() {
+        ResultSetMetaData product = null;
+        DBResultSetMetaData wrapper = null;
+        try {
+            product = resultSet.getMetaData();
+            wrapper = new DBResultSetMetaData(parent, product, name);
```

Comment:
update the constructor in `DBResultSetMetaData `to accept just `DBResultSet`, to be consistent with other classes that extend `AbstractParentWrapper`. Also, pass valid value for both `parent `and `name `variable, they are `null `at this point.